### PR TITLE
MODULES-5666: Implementing Rubocop in the module

### PR DESCRIPTION
**Pull request by david22swan**
Rubocop has been run.
Both Unit and Acceptance tests are running fine.
One Rubocop error disabled: Numeric Predicate error, that when changed as indicated causes test failures.

### CODE REVIEW

File: lib/puppet/reports/tagmail.rb [hunk 1]
```diff
@@ -177,12 +173,25 @@ def process(tagmail_conf_file = "#{Puppet[:confdir]}/tagmail.conf")
...
+                           rescue
+                             {}
+                           end
+    metrics['events'] = begin
+                          metrics['events'] || {}
+                        rescue
+                          {}
+                        end
+
+    if metrics['resources']['out_of_sync'] == 0 && metrics['resources']['changed'] == 0 && metrics['events']['audit'].nil? # rubocop:disable Style/NumericPredicate
```
Comment (tphoney) [inline lib/puppet/reports/tagmail.rb]: does metrics['resources']['out_of_sync'].zero? work ?

File: lib/puppet/reports/tagmail.rb [hunk 2]
```diff
@@ -201,44 +210,44 @@ def send(reports)
...
-            end
-          end
-        rescue => detail
-          message = "Could not send report emails through smtp: #{detail}"
-          Puppet.log_exception(detail, message)
-          raise Puppet::Error, message, detail.backtrace
-        end
-      else
-        begin
+    # Thread.new {
```
Comment (tphoney) [inline lib/puppet/reports/tagmail.rb]: Below seems to have changes substantially. Assuming the unit tests and acceptance tests work. I am ok with it.

File: lib/puppet/reports/tagmail.rb (hunk 1 above)
Comment (david22swan) [reply lib/puppet/reports/tagmail.rb]: Nope, causes a 'nil:NilClass' on the other two comparisons.

Comment (david22swan) [reply lib/puppet/reports/tagmail.rb]: other comparison****

Comment (david22swan) [reply lib/puppet/reports/tagmail.rb]: What I mean is that it causes 'nil:NilClass' errors on some comparisons.

Comment (hunner) [reply lib/puppet/reports/tagmail.rb]: ah, because `nil.zero?` doesn't work but `nil == 0` would.

File: lib/puppet/reports/tagmail.rb (hunk 2 above)
Comment (hunner) [reply lib/puppet/reports/tagmail.rb]: Looks like code was still indented for commented-out `Thread.new {` and reviewing with `?w=1` makes the change much more clear.

**Pull request merged**